src/main/docs/guide/httpServer/serverConfiguration/cors/annotationBasedCors.adoc

```diff
@@ -0,0 +1,26 @@
+Micronaut CORS configuration applies by default to all endpoints in the running application.
+
+As an alternative, <<corsConfiguration, CORS configuration>> can be applied in a more fine-grained manner to specific routes using the
+link:{api}/io/micronaut/http/server/cors/CrossOrigin.html[@CrossOrigin] annotation. This is applied to `@Controller` to apply the CORS configuration to all endpoints in the controller. Alternatively, the annotation can be applied to specific endpoints on a controller, for even more fine-grained control.
+
+The `@CrossOrigin` annotation maps with a one-to-one correspondence to application-wide link:{api}/io/micronaut/http/server/cors/CorsOriginConfiguration.html[CorsOriginConfiguration] configuration properties. To specify just an allowed origin, use `@CrossOrigin("https://foo.com")`. To specify additional configuration details, use  a combination of annotation attributes the same as you would for specifying global `CorsOriginConfiguration` properties.
+
+[source,java]
+----
+@CrossOrigin(
+	allowedOrigins = { "http://foo.com" ,"^http(|s):\\/\\/www\\.google\\.com$" },
```

Comment:
I think using regex here is problematic. Like this example, it would match http://foo.com and http://fooxcom. Not an issue when there's no subdomain, but if there is, it allows an attacker to hijack the domain: When http://www.foo.com is configured for cors, an attacker may register wwwxfoo.com, and get full cors access. This means it's easy with regex to accidentally make an insecure configuration.

(yes I know this is already the current behavior)

Reply:
would be ok if allowedOrigins defaulted to not use regex unless you enable it. And we will default to be disabled?

Reply:
yes that works

Reply:
agreed